### PR TITLE
fix: appset status not reflected in CR

### PR DESCRIPTION
**Pull request by jaideepr97**
Signed-off-by: Jaideep Rao <jaideep.r97@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

 /kind bug
> /kind chore
> /kind cleanup
> /kind failing-test
> /kind enhancement
> /kind documentation
> /kind code-refactoring

**What does this PR do / why we need it**:
This PR adds applicationSetController to `.status` of the Argo CD CR. This allows us to track the status/health of the appset controller workload in the CR, and lets us know if the controller is running as expected or pending if a pod couldn't come up. This is needed since it is a dependency for being able to set up monitoring alerts for all argo-cd workloads to be notified when they remain unavailable for too long

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes https://github.com/argoproj-labs/argocd-operator/issues/627

**How to test changes / Special notes to the reviewer**:

1. Start operator locally
2. Create basic Argo CD instance 
```
cat <<EOF | oc apply -f -
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: argocd
spec: {}
EOF

```

3. Verify that `.status.applicationSetController` is initially set to `Unknown`
```
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: example-argocd
status:
  ...
  applicationSetController: Unknown
```

4. Enable applicationSet:
```
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: example-argocd
spec:
  applicationSet: {}
```

5. Verify that `.status.applicationSetController` is initially set to `Running`
```
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: example-argocd
status:
  ...
  applicationSetController: Running
```

6. Set `.spec.applicationSet.image` to some invalid image and delete existing appset deployment 
```
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: example-argocd
spec:
  applicationSet:
    image: quay.io/argoproj/argocd@sha256:8283a9f06033c2377dc61b03daf49
```
`kubectl delete deployment example-argocd-applicationset-controller -n $NAMESPACE`

7.  Verify that `.status.applicationSetController` is now set to `Pending` as the new controller pod fails to come up due to invalid image name
```
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: example-argocd
status:
  ...
  applicationSetController: Pending
```

### CODE REVIEW

File: api/v1alpha1/argocd_types.go [hunk 1]
```diff
@@ -791,6 +791,15 @@ type ArgoCDStatus struct {
 	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="ApplicationController",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
 	ApplicationController string `json:"applicationController,omitempty"`
 
+	// ApplicationSetController is a simple, high-level summary of where the Argo CD applicationSet controller component is in its lifecycle.
+	// There are five possible ApplicationSetController values:
```
Comment (ishitasequeira) [inline api/v1alpha1/argocd_types.go:795]: nit: `four possible ApplicationSetController values`

File: api/v1alpha1/argocd_types.go [hunk 2]
```diff
@@ -791,6 +791,15 @@ type ArgoCDStatus struct {
...
+	// Pending: The Argo CD applicationSet controller component has been accepted by the Kubernetes system, but one or more of the required resources have not been created.
+	// Running: All of the required Pods for the Argo CD applicationSet controller component are in a Ready state.
+	// Failed: At least one of the  Argo CD applicationSet controller component Pods had a failure.
+	// Unknown: For some reason the state of the Argo CD applicationSet controller component could not be obtained.
```
Comment (ishitasequeira) [inline api/v1alpha1/argocd_types.go:799]: nit: remove `For some reason`

**Review by ishitasequeira — APPROVED** [review]
Except for few nits! PR LGTM!!

File: api/v1alpha1/argocd_types.go (hunk 2 above)
Comment (jaideepr97) [reply api/v1alpha1/argocd_types.go]: fixed in all comments, thanks

File: deploy/olm-catalog/argocd-operator/0.5.0/argoproj.io_argocds.yaml [hunk 3]
```diff
@@ -6292,4 +6292,4 @@ status:
     kind: ""
     plural: ""
   conditions: []
-  storedVersions: []
+  storedVersions: []
```
Comment (iam-veeramalla) [inline deploy/olm-catalog/argocd-operator/0.5.0/argoproj.io_argocds.yaml:6295]: Update to 0.5.0 should be removed

Comment (iam-veeramalla) [inline deploy/olm-catalog/argocd-operator/0.5.0/argoproj.io_argocds.yaml:6295]: Updates to 0.5.0 should be removed

File: deploy/olm-catalog/argocd-operator/0.5.0/argocd-operator.v0.5.0.clusterserviceversion.yaml [hunk 4]
```diff
@@ -1178,4 +1178,4 @@ spec:
   provider:
     name: Argo CD Community
   replaces: argocd-operator.v0.4.0
-  version: 0.5.0
+  version: 0.5.0
```
Comment (iam-veeramalla) [inline deploy/olm-catalog/argocd-operator/0.5.0/argocd-operator.v0.5.0.clusterserviceversion.yaml:1181]: Updates to 0.5.0 should be removed

**Review by iam-veeramalla — CHANGES_REQUESTED** [review]

File: deploy/olm-catalog/argocd-operator/0.5.0/argoproj.io_argocds.yaml (hunk 3 above)
Comment (jaideepr97) [reply deploy/olm-catalog/argocd-operator/0.5.0/argoproj.io_argocds.yaml]: done

File: deploy/olm-catalog/argocd-operator/0.5.0/argocd-operator.v0.5.0.clusterserviceversion.yaml (hunk 4 above)
Comment (jaideepr97) [reply deploy/olm-catalog/argocd-operator/0.5.0/argocd-operator.v0.5.0.clusterserviceversion.yaml]: done

**Review by iam-veeramalla — APPROVED** [review]

**Pull request merged**